File: src/database/DBGeoLoc.java
package database;

import model.GeoLoc;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

/**
 * DBGeoLoc
 * Handles all database functionality for GeoLocs.
 *
 * @author Dani Sander
 * @version 1.0
 * @since 20-05-15
 */

public class DBGeoLoc {

    private DBConnection dbConnection;
    private static DBGeoLoc instance;

    /**
     * Private constructor for singleton.
     *
     * @throws SQLException
     * @throws ClassNotFoundException
     */
    private DBGeoLoc() throws ClassNotFoundException, SQLException {
        dbConnection = DBConnection.getInstance();
    }

    /**
     * Singleton method for class.
     *
     * @return instance of class.
     * @throws SQLException
     * @throws ClassNotFoundException
     */
    public static DBGeoLoc getInstance() throws ClassNotFoundException, SQLException {
        if (instance == null) {
            instance = new DBGeoLoc();
        }

        return instance;
    }

    public GeoLoc getGeoLoc(long defaultDeliveryStopID) {
        ArrayList<GeoLoc> list;
        String sql = String.format("select * from GeoLoc where id = %d",
                defaultDeliveryStopID);
        list = (ArrayList<GeoLoc>) dbConnection.sendSQL(this, sql, "_formatGeoLoc");
        return list.get(0);
    }

    public ArrayList<GeoLoc> _formatGeoLoc(ResultSet rs) {
        ArrayList<GeoLoc> tableList = new ArrayList<>();
        try {
            while (rs.next()) {
                long tmp = rs.getLong("id");
                tableList.add(new GeoLoc(
                        rs.getLong("id"),
                        rs.getDouble("x"),
                        rs.getDouble("y")
                ));
            }
        } catch (SQLException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        return tableList;
    }
}

File: Database/DBGeoLoc.cs
using System;
using System.Collections.Generic;
using Model;
using System.Data;


namespace Database
{
    public class DBGeoLoc
    {
        private DBConnection DbConnection { get; set; }
        private static DBGeoLoc instance;

        /// <summary>
        /// Private singleton constructor.
        /// </summary>   
        private DBGeoLoc()
        {
            DbConnection = DBConnection.Instance;
        }

        /// <summary>
        /// Singleton method. Returns the instance of the class.
        /// </summary>
        /// <returns>Instance of class.</returns>
        public static DBGeoLoc Instance
        {
            get
            {
                if (instance == null)
                {
                    instance = new DBGeoLoc();
                }

                return instance;
            }

        }

        /// <summary>
        /// Returns a geoLoc...
        /// </summary>
        /// <param name="defaultDeliveryStopID"></param>
        /// <returns></returns>
        public GeoLoc getGeoLoc(long defaultDeliveryStopID)
        {
            List<GeoLoc> list;
            String sql = String.Format("select * from GeoLoc where id = {0}",
                    defaultDeliveryStopID);
            list = DbConnection.SendSQL<GeoLoc>(sql, ConvertToGeoLoc);
            return list[0];
        }

        /// <summary>
        /// Return a list of all GeoPoints
        /// </summary>
        /// <returns>GeoPoints</returns>
        public List<GeoLoc> GetGeoLocs()
        {
            String sql = String.Format("select * from GeoLoc");
            return DbConnection.SendSQL<GeoLoc>(sql, ConvertToGeoLoc);
        }

        /// <summary>
        /// Takes the ReulstSet from database and returns a list of all GeoLocs.
        /// </summary>
        /// <param name="rs"></param>
        /// <returns></returns>
        private List<GeoLoc> ConvertToGeoLoc(IDataReader dataSet)
        {
            List<GeoLoc> tableList = new List<GeoLoc>();
            while (dataSet.Read())
            {
                tableList.Add(new GeoLoc(
                    dataSet.GetInt64(0),
                    dataSet.GetDouble(2),
                    dataSet.GetDouble(1)
                ));
            }

            return tableList;
        }
    }
}